pep-0634.rst

```diff
@@ -0,0 +1,706 @@
+PEP: 634
+Title: Structural Pattern Matching: Specification
+Version: $Revision$
+Last-Modified: $Date$
+Author: Brandt Bucher <brandtbucher@gmail.com>,
+        Guido van Rossum <guido@python.org>
+BDFL-Delegate:
+Discussions-To: Python-Dev <python-dev@python.org>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 12-Sep-2020
+Python-Version: 3.10
+Post-History:
+Replaces: 622
+Resolution:
+
+
+Abstract
+========
+
+TODO: Do we use "the subject matches the pattern" or "the pattern
+matches the subject"?
+
+This PEP provides the technical specification for the ``match``
+statement.  It replaces PEP 622, which is hereby split in three parts:
+
+- PEP 634: Specification
+- PEP 635: Motivation and Rationale
+- PEP 636: Tutorial
+
+This PEP is intentionally devoid of commentary; all explanations of
+design choices are in PEP 635.  First-time readers are encouraged to
+start with PEP 636, which provides a gentler introduction to the
+concepts, syntax and semantics of patterns.
+
+
+Syntax and Semantics
+====================
+
+TODO: Should we show the lookaheads in the grammar?  They are a bit
+ugly.  Maybe as a compromise show them in the appendix but not in the
+main text?
```

Comment:
Yep, good idea. I'll strip them out.

Reply:
We still need the lookaheads that distinguish captures from wildcards. I am putting those back in.

Reply:
Hm, okay. It's strange because the wildcards aren't actually part of the grammar (they're just treated differently during compilation).

The section for wildcards is so short. Shall we just combine it with capture patterns, and then we don't need to worry about the lookahead in the grammar?